projects/aca-playwright-shared/src/page-objects/components/dialogs/link-rules.component.ts

```diff
@@ -0,0 +1,64 @@
+/*!
+ * Copyright © 2005-2024 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Alfresco Example Content Application
+ *
+ * This file is part of the Alfresco Example Content Application.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail. Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Locator, Page, expect } from '@playwright/test';
+import { BaseComponent } from '../base.component';
+
+export class LinkRulesDialog extends BaseComponent {
+  private static rootElement = 'aca-rule-set-picker';
+
+  constructor(page: Page) {
+    super(page, LinkRulesDialog.rootElement);
+  }
+
+  public cancelButton = this.getChild('[data-automation-id="content-node-selector-actions-cancel"]');
+  public selectFolderButton = this.getChild('button', { hasText: ' Select folder ' });
+  public emptyLinkRules = this.getChild('.adf-empty-content__title');
+  getOptionLocator = (optionName: string): Locator => this.page.locator('.mat-select-panel .mat-option-text', { hasText: optionName });
```

Comment:
Some of those material classes might not be available as soon as angular 15 migration is merged

Reply:
Okay I will try to update them